Instructions/movement_instruction.py

```diff
@@ -0,0 +1,32 @@
+from collections import deque
+
+# Ours
+from ..Tasks.movement_task import MovementTask
+from instruction_base import InstructionBase
+from ..Utilities import constants as c
+
+class MovementInstruction(InstructionBase):
+
+    def __init__(self, data):
+        super(MovementInstruction, self).__init__()
+        # The data here will eventually be the raw binary data that
+        # has traveled over the network. The decoding may be more
+        # involved than what is shown here.
+        (self.x, self.y, self.z) = data
+        
+        self.movementTask = None
+
+    def get_task(self, drone):
+        queue = deque()
+        if self.x != 0:
+            queue.append((c.BACKWARD if self.x > 0 else c.FORWARD, abs(self.x)))
```

Comment:
Does not follow DRY principle. Consider using code analogous to the following:

direction_map = {
    'x':  (c.FORWAD, c.BACKWARD),
    'y': (c.LEFT, c.RIGHT),
    'z': (c.UP, c.DOWN)
}
positive_vector_index = 0
negative_vector_index = 1
for direction, vectors in direction_map.items():
    value = getattr(self, direction)
    if value != 0:
        queue.append((vectors[postive_vector_index if value > 0 else negative_value_index], abs(value))